ddl/column.go

```diff
@@ -1556,6 +1557,53 @@ func updateChangingObjState(changingCol *model.ColumnInfo, changingIdxs []*model
 	}
 }
 
+func needCheckLatin1Convert(oldCol, newCol *model.ColumnInfo) bool {
+	if oldCol.GetCharset() != newCol.GetCharset() {
+		return oldCol.GetCharset() == charset.CharsetLatin1
+	}
+	return false
+}
+
+func (w *worker) checkLatin1Convert(schema, t model.CIStr, oldCols []*model.ColumnInfo, newCols []*model.ColumnInfo) (err error) {
+	sctx, err := w.sessPool.get()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	sctx.GetSessionVars().StmtCtx.TruncateAsWarning = false
+	sctx.GetSessionVars().StmtCtx.IgnoreTruncate = false
+	defer w.sessPool.put(sctx)
+
+	var buf strings.Builder
+	buf.WriteString("select * from %n.%n where ")
```

Comment:
```suggestion
	buf.WriteString("select 1 from %n.%n where ")
```

Reply:
We need the result for generate error msg.